pkg/operation/botanist/component/vpnseedserver/vpn_seed_server.go

```diff
@@ -0,0 +1,645 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vpnseedserver
+
+import (
+	"context"
+	"fmt"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/managedresources"
+	"github.com/gogo/protobuf/types"
+
+	"istio.io/api/networking/v1beta1"
+	networkingv1beta1 "istio.io/client-go/pkg/apis/networking/v1beta1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	// VpnSeedServerTLSAuth is the name of seed server tlsauth Secret.
+	VpnSeedServerTLSAuth = "vpn-seed-server-tlsauth"
+	// VpnSeedServerDH is the name of seed server DH Secret.
+	VpnSeedServerDH = "vpn-seed-server-dh"
+	// DeploymentName is the name of vpn seed server deployment.
+	DeploymentName = v1beta1constants.DeploymentNameVPNSeedServer
+	// EnvoyConfig is the name of the vpn seed server envoy configuration.
+	EnvoyConfig = "vpn-seed-server-envoy-config"
+	// MountPathVpnSeedServer contains the mount path for the vpn seed server secret.
+	MountPathVpnSeedServer = "/srv/secrets/vpn-server"
+	// MountPathTLSAuth contains the mount path for the tlsauth secret.
+	MountPathTLSAuth = "/srv/secrets/tlsauth"
+	// MountPathDH contains the mount path for the dh secret.
+	MountPathDH = "/srv/secrets/dh"
+)
+
+// VPNSeedServer contains functions for a vpn-seed-server deployer.
+type VPNSeedServer interface {
+	component.DeployWaiter
+
+	// SetSecrets sets the secrets.
+	SetSecrets(Secrets)
+}
+
+// Secrets is collection of secrets for the vpn-seed-server.
+type Secrets struct {
+	// TLSAuth is a secret containing the TLSAuth certificate.
+	TLSAuth component.Secret
+	// Server is a secret containing the server certificate and key.
+	Server component.Secret
+	// DH is a secret containing the DH key.
+	DH component.Secret
```

Comment:
Nit: I would name this field explicit, i.e., `DiffieHellmanKey`

Reply:
Renamed the field as suggested.